.github/workflows/build.yml

```
@@ -0,0 +1,45 @@
+name: Build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - trunk
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+      - run: ./gradlew build
+      - uses: actions/upload-artifact@v4
+        with:
+          name: build-out-jars
+          path: build/libs/*.jar
+
+  publish:
+    needs: build
+    if: github.repository == 'JakeWharton/dalvik-dx'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+      - uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: true
+      - name: Publish Artifacts
+        run: ./gradlew publish
+        if: github.ref == 'refs/heads/trunk'
```

```suggestion
      - run: ./gradlew publish
        if: ${{ github.ref == 'refs/heads/trunk' && github.repository == 'JakeWharton/dalvik-dx' }}
```
No need to build twice